File: bench/dotnet/errors-deep.cs
namespace Bench;

static class ErrorsDeep
{
    const int DEPTH = 50;

    static int Descend(int d, int i)
    {
        if (d == 0) throw new BenchError(i % 100);
        return 1 + Descend(d - 1, i);
    }

    public static void Run(int n)
    {
        const long MOD = 1000000007;
        long acc = 0;
        for (int i = 0; i < n; i++)
        {
            try { Descend(DEPTH, i); }
            catch (BenchError e) { acc += e.V; }
        }
        Console.WriteLine(acc % MOD);
    }
}

File: bench/elixir/errors-deep.ex
defmodule BenchErrorDeep do
  defexception [:v, message: "bench error"]
end

defmodule Berrorsdeep do
  def descend(0, i), do: raise(BenchErrorDeep, v: rem(i, 100))
  def descend(d, i), do: 1 + descend(d - 1, i)

  def main do
    n = String.to_integer(System.get_env("BENCH_N") || "50000")
    md = 1000000007
    depth = 50

    acc =
      Enum.reduce(0..(n - 1), 0, fn i, acc ->
        try do
          descend(depth, i)
        rescue
          e in BenchErrorDeep -> acc + e.v
        end
      end)

    IO.puts(rem(acc, md))
  end
end
